tests/integration/query.ts

```diff
@@ -50,3 +58,110 @@ it('universal autocomplete sets search intents', async () => {
   expect(statefulCore.state.query.searchIntents).toEqual(['NEAR_ME']);
 });
 
+
+describe('ensure correct results from latest request', () => {
+  const mockedCore: any = {
+    verticalAutocomplete: jest.fn(
+      async (request: VerticalAutocompleteRequest) => {
+        const waitTime = request.input?.length;
+        return new Promise(res => setTimeout(() => res(
+          { results: [ {value: request.input} ]}), waitTime));
+      }),
+    universalAutocomplete: jest.fn(
+      async (request: UniversalAutocompleteRequest) => {
+        const waitTime = request.input?.length;
+        return new Promise(res => setTimeout(() => res(
+          { results: [{ value: request.input }] }), waitTime));
+      }),
+    verticalSearch: jest.fn( async (request: VerticalSearchRequest) => {
+      const waitTime = request.query?.length;
+      return new Promise(res => setTimeout(() => res(
+        { verticalResults: { results: [request.query] } }), waitTime));
+    }),
+    universalSearch: jest.fn( async (request: UniversalSearchRequest) => {
+      const waitTime = request.query?.length;
+      return new Promise(res => setTimeout(() => res(
+        { verticalResults: [{ results: [request.query] }] }), waitTime));
+    })
+  };
+  const stateManager = new ReduxStateManager();
+  const statefulCore = new StatefulCore(mockedCore, stateManager);
+  const queries = ['long request', 'short'];
+  const dispatchEventSpy = jest.spyOn(stateManager, 'dispatchEvent');
+  const addListenerSpy = jest.spyOn(stateManager, 'addListener');
+
+  beforeEach(() => {
+    statefulCore.setVerticalKey('someKey');
+    jest.clearAllMocks();
+  });
+
+  it('vertical autocomplete get latest request results', async () => {
+    statefulCore.setQuery(queries[0]);
+    const firstResponsePromise = statefulCore.executeVerticalAutoComplete();
+    statefulCore.setQuery(queries[1]);
+    const secondResponsePromise = statefulCore.executeVerticalAutoComplete();
+
+    await Promise.all([firstResponsePromise, secondResponsePromise]);
```

Comment:
Rather than waiting for both to resolve, could we use jest.advanceTimersByTime? For example, if we have one query take 10ms and another take 100ms, we could run `jest.advanceTimersByTime(10)` and check that shorter query has resolved and that the longer query has not yet resolved. And then we can run `jest.runAllTimers` so that the longer query is resolved. This way we know for sure the resolution order, and we also get the added benefit of not slowing down our tests by waiting for the timeouts

Reply:
oh that's cool. didn't know about that, I will try something out with it! In that case, it wouldn't matter much if we make the time gap to be small (5ms & 12ms) since this would just advance the time.